tooling/ssa_fuzzer/fuzzer/src/base_context.rs

```diff
@@ -0,0 +1,374 @@
+    brillig_arrays: Vec<Array>,
+    /// Whether the context is constant execution
+    is_constant: bool,
+}
+
+impl FuzzerContext {
+    /// Creates a new fuzzer context with the given type
+    pub(crate) fn new(type_: Type) -> Self {
+        let mut acir_builder = FuzzerBuilder::new_acir();
+        let mut brillig_builder = FuzzerBuilder::new_brillig();
+        acir_builder.insert_variables(type_.clone());
+        brillig_builder.insert_variables(type_.clone());
+        let mut acir_ids = vec![];
+        let mut brillig_ids = vec![];
+        // by default private variables ids are indexed from 0 to NUMBER_OF_VARIABLES_INITIAL - 1
+        for i in 0..config::NUMBER_OF_VARIABLES_INITIAL {
+            acir_ids.push(i);
+            brillig_ids.push(i);
+        }
+        Self {
+            acir_builder,
+            brillig_builder,
+            acir_ids,
+            brillig_ids,
+            acir_arrays: vec![],
+            brillig_arrays: vec![],
+            is_constant: false,
+        }
+    }
+
+    /// Creates a new fuzzer context with the given values for a constant folding checking
+    pub(crate) fn new_constant(values: Vec<impl Into<FieldElement>>, type_: Type) -> Self {
+        let mut acir_builder = FuzzerBuilder::new_acir();
+        let mut brillig_builder = FuzzerBuilder::new_brillig();
+        acir_builder.set_type(type_.clone());
+        brillig_builder.set_type(type_.clone());
+        let mut acir_ids = vec![];
+        let mut brillig_ids = vec![];
+        for value in values {
+            let field_element = value.into();
+            acir_ids.push(id_to_int(acir_builder.insert_constant(field_element)));
+            brillig_ids.push(id_to_int(brillig_builder.insert_constant(field_element)));
+        }
+        Self {
+            acir_builder,
+            brillig_builder,
+            acir_ids,
+            brillig_ids,
+            acir_arrays: vec![],
+            brillig_arrays: vec![],
+            is_constant: true,
+        }
+    }
+
+    /// Creates a new array from a vector of indices of variables
+    /// Skips non-presented variables
+    pub(crate) fn insert_array(&mut self, elements: Vec<u32>) {
+        let mut acir_values_ids = vec![];
+        let mut brillig_values_ids = vec![];
+        for elem in elements {
+            let acir_len = self.acir_ids.len();
+            let brillig_len = self.brillig_ids.len();
+            acir_values_ids.push(elem % acir_len as u32);
+            brillig_values_ids.push(elem % brillig_len as u32);
+        }
+        let acir_len = acir_values_ids.len();
+        let brillig_len = brillig_values_ids.len();
+        let acir_array = self.acir_builder.insert_make_array(acir_values_ids);
+        let brillig_array = self.brillig_builder.insert_make_array(brillig_values_ids);
+        self.acir_arrays.push(Array { id: acir_array, length: acir_len as u32 });
+        self.brillig_arrays.push(Array { id: brillig_array, length: brillig_len as u32 });
```

Comment:
```suggestion
    pub(crate) fn insert_array(&mut self, elements: &[u32]) {
        let mut acir_values_ids = vec![];
        let mut brillig_values_ids = vec![];
        let acir_len = self.acir_ids.len();
        let brillig_len = self.brillig_ids.len();
        for elem in elements {
            acir_values_ids.push(elem % acir_len as u32);
            brillig_values_ids.push(elem % brillig_len as u32);
        }
        let acir_array = self.acir_builder.insert_make_array(acir_values_ids);
        let brillig_array = self.brillig_builder.insert_make_array(brillig_values_ids);
        self.acir_arrays.push(Array { id: acir_array, length: elements.len() as u32 });
        self.brillig_arrays.push(Array { id: brillig_array, length: elements.len() as u32 });
```
Most of the time we pass `&[_]` instead of `Vec<_>` unless you really want to pass ownership. You don't have to, just a bit more general as it works with arrays too.